### PR TITLE
Feat/typescript vue apollo client id in config

### PR DESCRIPTION
**Pull request by swunderlich**
## Description

Introduces the possibility to add a clientId

Related #362 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

I extended the existing test of the composition function generation

## Checklist:

- [x] I have followed the
      [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the
      style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by oikalyptus — APPROVED** [review]

**Review by saihaj — APPROVED** [review]

**Comment by saihaj** [comment]
hey @swunderlich can you sync this branch with `main` that way we can run the CI and make sure tests run

**Comment by swunderlich** [comment]
@saihaj done

File: packages/plugins/typescript/vue-apollo/src/visitor.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ export interface VueApolloPluginConfig extends ClientSideBasePluginConfig {
   vueApolloComposableImportFrom: 'vue' | '@vue/apollo-composable' | string;
   vueCompositionApiImportFrom: 'vue' | '@vue/apollo-composable' | string;
   addDocBlocks: boolean;
+  clientId: string | null;
```
Comment (saihaj) [inline packages/plugins/typescript/vue-apollo/src/visitor.ts:19]: we don't want a required prop given that can be a breaking changes for users.
```suggestion
  clientId?: string | null;
```

**Review by saihaj — APPROVED** [review]
looks great. Thanks!

**Pull request merged**